src/ape/managers/chain.py

```diff
@@ -0,0 +1,143 @@
+from typing import List, Optional
+
+from dataclassy import dataclass
+
+from ape.api import BlockAPI, ProviderAPI, TestProviderAPI
+from ape.exceptions import ChainError, ProviderNotConnectedError, UnknownSnapshotError
+from ape.types import SnapshotID
+
+from .networks import NetworkManager
+
+
+@dataclass
+class BlockContainer:
+    """
+    A list of blocks on the chain.
+
+    Usages example::
+
+        from ape import chain
+
+        latest_block = chain.blocks[-1
+
+    """
+
+    _provider: ProviderAPI
+
+    def __getitem__(self, block_number: int) -> BlockAPI:
+        """
+        Get a block by number.
+
+        Args:
+            block_number (int): The number of the block to get.
+
+        Returns:
+            :class:`~ape.api.providers.BlockAPI`
+        """
+        if block_number < 0:
+            latest_block = self._provider.get_block("latest").number
+            block_number = latest_block + 1 + block_number
+
+        return self._provider.get_block(block_number)
+
+    def __len__(self) -> int:
+        """
+        The number of blocks in the chain.
+
+        Returns:
+            int
+        """
+
+        return self._provider.get_block("latest").number
+
+
+@dataclass
+class ChainManager:
+    """
+    A manager for controlling and manipulating development blockchains
+    and querying chain-data.
+    """
+
+    _networks: NetworkManager
+    _snapshots: List[SnapshotID] = []
+
+    blocks: BlockContainer = None  # type: ignore
```

Comment:
do you even have to put this here with `__post_init__`?
```suggestion
    blocks: BlockContainer  # type: ignore
```

Reply:
Yeah it seems like it does